MAGECLOUD-3057 remove Xdebug hack (#6070)
magento/devdocs: guides/v2.2/cloud/howtos/debug.md

  - Setup
---

> Xdebug is an extension for debugging your PHP. The following explains how to configure Xdebug and PhpStorm to debug in your local environment. You can use any IDE of your choice. See the vendor documentation for those applications for further configuration information.

Reviewer: ```suggestion
Xdebug is an extension for debugging your PHP. The following explains how to configure Xdebug and PhpStorm to debug in your local environment. You can use the IDE of your choice. See the vendor documentation for those applications for further configuration information.
```

> To set up Xdebug, you need to [configure](#configure-xdebug) a file in your Git repository, configure your IDE PhpStorm, and set up port forwarding. You can configure settings in the `magento.app.yaml` file. After editing, you can push the Git changes across all Starter environments and Pro Integration environments to enable Xdebug. To push these settings to Pro plan Staging and Production environments, you must enter a ticket.

Reviewer: ```configure your IDE PhpStorm``` -- Should PhpStorm be removed, or something is missing?

> Once configured, you can debug [CLI commands](#debugcli), [web requests](#webrequests), [code](#code). Remember, all {{site.data.var.ece}} environments are read-only. You need to pull code to your local development environment to perform debugging. For Pro Staging and Production environments, we include [additional instructions](#pro-debug) for Xdebug.

Reviewer: ```suggestion
Once configured, you can debug [CLI commands](#debugcli), [web requests](#webrequests), and [code](#code). Remember, all {{site.data.var.ece}} environments are read-only. You need to pull code to your local development environment to perform debugging. For Pro Staging and Production environments, we include [additional instructions](#pro-debug) for Xdebug.
```

## Requirements {#usexdebug}

> To run and use Xdebug, you need the SSH URL for the environment. You can locate the information through the [Project Web Interface]({{ page.baseurl }}/cloud/project/projects.html) or your spreadsheet.

Reviewer: ```suggestion
To run and use Xdebug, you need the SSH URL for the environment. You can locate the information through the [Project Web Interface]({{ page.baseurl }}/cloud/project/projects.html) or your Cloud Onboarding UI.
```

Reviewer: Onboarding spreadsheet not used anymore.  Now use Cloud Onboarding UI -- that's the name that CSM and TAMs use to refer to the interface created by Rainbow team.

### Enable Xdebug in your environment {#enable}

To enable Xdebug for your project, add `xdebug` to the `runtime:extensions` section of the `.magento.app.yaml` file.

> You can enable Xdebug directly to all Starter environments and Pro Integration environments. For Pro Staging and Production, you need to update this file and enter a [Support ticket]({{ page.baseurl }}/cloud/trouble/trouble.html) to have it enabled. We enable Xdebug onto those environments for you.

Reviewer: ```suggestion
You can enable Xdebug directly to all Starter environments and Pro Integration environments. For Pro Staging and Production, you need to update this file and enter a [Support ticket]({{ page.baseurl }}/cloud/trouble/trouble.html) to have it enabled. We enable Xdebug on those environments for you.
```

1. Click the **+** to add a server configuration. The project name is in grey at the top.

1. Configure the following settings for the new server configuration:

   -  **Name**—enter the same as the hostname. This value is used in and must match the value for `PHP_IDE_CONFIG` variable in [Debug CLI commands](#debugcli).
   -  **Host**—Enter `localhost`.
   -  **Port**—Enter `80`.
   -  **Debugger**—Select `Xdebug`.

1. Select **Use path mappings**. In the _File/Directory_ pane, the root of the project for the `serverName` displays.

1. In the **Absolute path on the server** column, click ![Edit]({{ site.baseurl }}/common/images/install_docker_php-storm-edit.png){:width="15px"} (**Edit**) and add a setting based on the environment:

   -  For all Starter environments and Pro Integration environments, the remote path is `/app`.
   -  For Pro Staging and Production environments:

      -  Production: `/app/<project_code>/`
      -  Staging:  `/app/<project_code>_stg/`

1. Change the Xdebug port to 9001 in the **Languages & Frameworks** > **PHP** > **Debug** > **Xdebug** > **Debug Port** panel.

1. Click **Apply**.

### Set up port forwarding {#port}

> You need to set up port forwarding. This is necessary to map the XDEBUG connection from the server to your local system to PHPSTORM and Xdebug locally can track along with it

Reviewer: Last part of sentence "and Xdebug locally can track along with it"  -- not sure what this means. Maybe missing comma, or maybe "and" is "so that"?  Also, missing period at end of 2nd sentence.

Author: Yeah, yuck.

To set up port forwarding (SSH tunneling) on Windows, you must configure your Windows terminal application. For this example, we walk through creating an SSH tunnel using [Putty](http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html). You can use other applications such as Cygwin. For more information on other applications, see the vendor documentation provided with those applications.

{:.procedure}
To set up an SSH tunnel on Windows using Putty:

1. If you have not already done so, download [Putty](http://www.chiark.greenend.org.uk/~sgtatham/putty/download.html).

1. Start Putty.

1. In the Category pane, click **Session**.

1. Enter the following information:

>    -  **Hostname (or IP address)** field: Enter your Cloud server's [SSH URL]({{ page.baseurl }}/cloud/env/environments-ssh.html)

Reviewer: ```suggestion
   -  **Hostname (or IP address)** field: Enter the [SSH URL]({{ page.baseurl }}/cloud/env/environments-ssh.html) for your Cloud server
```

1. In the _Category_ pane, click **Connection** > **SSH** > **Tunnels**.

1. Enter the following information:

   -  **Source port** field: Enter `9000`
   -  **Destination** field: Enter `127.0.0.1:9000`
   -  Click **Remote**

1. Click **Add**.

   ![Create an SSH tunnel in Putty]({{ site.baseurl }}/common/images/cloud-xdebug_putty-tunnels.png){:width="350px"}

1. In the _Category_ pane, click **Session**.

1. In the **Saved Sessions** field, enter a name for this SSH tunnel.

1. Click **Save**.

   ![Save your SSH tunnel]({{ site.baseurl }}/common/images/cloud-xdebug_putty-session-save.png){:width="350px"}

1. To test the SSH tunnel, click **Load**, then click **Open**.

   If an "unable to connect" error displays, verify all of the following:

   -  All Putty settings are correct
   -  You are running Putty on the machine on which your private {{site.data.var.ece}} SSH keys are located

### Configure Pro Staging and Production {#pro}

To complete configuration for Pro plan Staging and Production environments, you must enter a [Support ticket]({{ page.baseurl }}/cloud/trouble/trouble.html) to have Xdebug enabled and configured in Staging and Production environments.

> We enable Xdebug in the environment. Be aware, this requires a redeployment of Staging and Production.

Reviewer: ```suggestion
We enable Xdebug in the environment. Be aware that this is a configuration change that requires us to redeploy your Staging and Production environments.
```

You need the following:

> -  SSH commands for accessing the environments. You can get this information, through the [Project Web Interface]({{ page.baseurl }}/cloud/project/projects.html) and your project spreadsheet.

Reviewer: ```suggestion
-  SSH commands for accessing the environments. You can get this information, through the [Project Web Interface]({{ page.baseurl }}/cloud/project/projects.html) or you Cloud Onboarding UI.
```

-  The `xdebug_key` value we set when configuring the Staging and Pro environments

{:.procedure}

> To set up an SSH tunnel to Staging or Production environment:

Reviewer: ```suggestion
To set up an SSH tunnel to a Staging or Production environment:
```
